app/src/org/commcare/provider/IdentityCalloutHandler.java

```diff
@@ -109,8 +110,31 @@ private static boolean processRegistrationReponse(FormDef formDef,
                                                       Hashtable<String, Vector<TreeReference>> responseToRefMap) {
         RegistrationResult registrationResult = intent.getParcelableExtra(IdentityResponseBuilder.REGISTRATION);
         String guid = registrationResult.getGuid();
+
         storeValueFromCalloutInForm(formDef, responseToRefMap, intentQuestionRef, REF_GUID, guid);
-        IntentCallout.setNodeValue(formDef, intentQuestionRef, guid);
+
+        int numOfTemplatesStored = 0;
+        int numOfTemplates = 0;
+        for (Map.Entry<BiometricIdentifier, byte[]> template : registrationResult.getTemplates().entrySet()) {
+            boolean success = storeValueFromCalloutInForm(formDef, responseToRefMap,
+                    intentQuestionRef,
+                    template.getKey().getCalloutResponseKey(),
+                    BiometricUtils.convertTemplateToBase64String(template.getKey(), template.getValue()));
+            if (success) {
+                numOfTemplatesStored++;
+            }
+            numOfTemplates++;
+        }
+
+        String result;
+        if (registrationResult.getTemplates().isEmpty()) {
+            result = guid;
+        } else {
+            result = Localization.get("intent.callout.biometrics.capture.result",
+                    new String[]{String.valueOf(numOfTemplatesStored), String.valueOf(numOfTemplates)});
+        }
```

Comment:
My reasoning here is to provide some feedback to the user in terms of the number of biometrics that was successfully captured (and stored). It can be that we scan 4 fingers but the callout handler is only able to store 2 due to a misconfiguration in the app. But not a hard requirement, I'm happy to drop this and maybe wait for feedback.

Reply:
Got it. That makes sense to me although I am worried about supporting 2 different return formats to the callout question.  I am thinking about case exports where some fields might have guid and some fields this string which might be confusing to interpret for the data systems incorporated by a project.

That said I do like the idea of telling to user whether everything was saved correctly or not. How about we just say `All data stored successfully` or `Failed to store some data`  to indicate failure vs success in both templates and non templates use cases here.

Reply:
756586fa289c69c28791b4d7b083813d3d5a5daf